crates/cli-flags/src/lib.rs

```diff
@@ -81,6 +81,21 @@ wasmtime_option_group! {
         /// Configure attempting to initialize linear memory via a
         /// copy-on-write mapping (default: yes)
         pub memory_init_cow: Option<bool>,
+
+        /// The maximum number of WebAssembly instances which can be created.
+        pub total_core_instances: Option<u32>,
+
+        /// The maximum number of WebAssembly components which can be created.
+        pub total_component_instances: Option<u32>,
+
+        /// The maximum number of WebAssembly memories which can be created.
+        pub total_memories: Option<u32>,
+
+        /// The maximum number of WebAssembly tables which can be created.
+        pub total_tables: Option<u32>,
+
+        /// The maximum number of WebAssembly stacks which can be created.
+        pub total_stacks: Option<u32>,
```

Comment:
Another option here, since all of these default to the same number currently, is to have a single flag `--total-pool-slots` or the like that sets all of these together. But maybe someone wants to control these in a bit more fine-grained way, so I went down this route instead.